### PR TITLE
ci: build Docker image

### PR DESCRIPTION
**Pull request by eddieantonio**

### CODE REVIEW

**Comment by eddieantonio** [comment]
My current problems with this PR:

 1. ~~I can't pull the public image <https://github.com/orgs/UAlbertaALTLab/packages/container/package/itwewina.altlab.app> anonymously~~. **EDIT**: okay, apparently I _can_ today????
 2. WHEN should the latest package be built, tagged, and uploaded? I thought on push to master. Does this seem okay?

Thoughts? @nienna73 @andrewdotn @dwhieb ideas?

**Comment by nienna73** [comment]
Yay for GHCR!

I think having it update on push to master makes sense. That's when the server rebuilds the site anyway, right?

**Comment by eddieantonio** [comment]
> I think having it update on push to master makes sense. That's when the server rebuilds the site anyway, right?

Yes, I intend for this to push the Docker image ONLY on commits pushed to the default branch.

**Comment by eddieantonio** [comment]
I think I finally have this working right. On pushes to `master`:

 1. the unit tests, the integration tests, and the **Docker image build** occur simulataneously. The Docker image is pushed with a tag matching the current git SHA
 2. once the tests pass and the image is built, `trigger-deployment` renames the uploaded Docker image, tagging it with `latest`
 3. A request is sent to `deploy.altlab.app`; the new deployment script pulls the image (instead of buidling it!) and redeploys!

File: .github/workflows/test-and-deploy.yml [hunk 1]
```diff
@@ -187,9 +213,26 @@ jobs:
...
+        password: ${{ secrets.GHCR_PERSONAL_ACCESS_TOKEN }}
+    - name: Cache Docker layers
+      uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+    - name: Pull existing image
+      run: docker pull ghcr.io/ualbertaaltlab/itwewina.altlab.app:${{ github.sha }}
+    - name: Tag previously successfully created image as latest
+      run: docker tag ghcr.io/ualbertaaltlab/itwewina.altlab.app:${{ github.sha }} ghcr.io/ualbertaaltlab/itwewina.altlab.app:latest
+    - name: Update the latest tag
+      run : docker push ghcr.io/ualbertaaltlab/itwewina.altlab.app:latest
```
Comment (eddieantonio) [inline .github/workflows/test-and-deploy.yml:235]: Is there a better way of retagging the image _without_ pulling the image and running `docker tag` and `docker push`?

**Review by nienna73 — APPROVED** [review]
I think this looks good. I don't have an answer to your question, though :/

**Comment by eddieantonio** [comment]
> I think this looks good. I don't have an answer to your question, though :/

I'm reading up on it and it sounds annoying... but possible. 

https://dille.name/blog/2018/09/20/how-to-tag-docker-images-without-pulling-them/

I'll save it for a future PR though 😅

**Review by andrewdotn — APPROVED** [review]

**Comment by eddieantonio** [comment]
I'll merge this Monday morning... just in case 😅

**Pull request merged**